sklearn/model_selection/_validation.py

```diff
@@ -895,8 +957,10 @@ def _incremental_fit_estimator(estimator, X, y, classes, train, test,
         else:
             estimator.partial_fit(X_partial_train, y_partial_train,
                                   classes=classes)
-        train_scores.append(_score(estimator, X_train, y_train, scorer))
-        test_scores.append(_score(estimator, X_test, y_test, scorer))
+        train_scores.append(_score(estimator, X_train, y_train,
+                                   {'score': scorer})['score'])
```

Comment:
I think it's a bit ironic that all the user-facing functions change return types depending on input, but none of the internal ones do ....